tracee-rules/output.go

```diff
@@ -44,24 +47,53 @@ func setupOutput(resultWriter io.Writer, clock Clock, webhook string) (chan type
 			}
 
 			if webhook != "" {
-				payload, err := prepareJSONPayload(res)
-				if err != nil {
-					log.Printf("error preparing json payload for %v: %v", res, err)
-					continue
+				if err := sendToWebhook(res, webhook, webhookTemplate, realClock{}); err != nil {
+					log.Println(err)
 				}
-				resp, err := http.Post(webhook, "application/json", strings.NewReader(payload))
-				if err != nil {
-					log.Printf("error calling webhook for %v: %v", res, err)
-					continue
-				}
-				resp.Body.Close()
 			}
 		}
 	}()
 	return out, nil
 }
 
-func prepareJSONPayload(res types.Finding) (string, error) {
+func sendToWebhook(res types.Finding, webhook string, webhookTemplate string, clock Clock) error {
+	var payload string
+
+	switch {
+	case webhookTemplate != "":
+		t, err := template.New(filepath.Base(webhookTemplate)).
+			Funcs(map[string]interface{}{
+				"unixToRFC3339": func(unixTs float64) string {
```

Comment:
also potentially for a future PR, but would be easy enough to plug https://github.com/Masterminds/sprig here

Reply:
Could you share an example of how you would use it in this case? I actually tried before I wrote a custom func and was unable to go from float64 -> rfc3339 date.

Reply:
On a side note: Why is timestamp float64 to begin with? Shouldn't it be an int64?

Reply:
> Could you share an example

{{ now | date "2006-01-02 15:04:05" }}

Reply:
> Why is timestamp float64 to begin with? Shouldn't it be an int64?

as I wrote in https://github.com/aquasecurity/tracee/pull/582#discussion_r584263000, this is not a unix timestamp

Reply:
Thanks, I understand it now. I updated it here https://github.com/aquasecurity/tracee/pull/582/commits/7ac6a9d705f9181964cab6718d69a80e75ed526b - note I still use a custom func because:

1. Besides time, we don't have any other func we are using. So we can avoid an external dependency. 
2. Its testable as a format output. With using the external library, we will have to forgo that control as our template output will include dynamically generated time which cannot be injected.

Reply:
that's fine, it was just a suggestion. I did think however that we would want to add it at some point regardless of the time field here, as a general "stdlib" available to template authors. but we can postpone this